### PR TITLE
First draft alw-integration

### PR DESCRIPTION
**Pull request by martind260**
---
name: Pull request
about: Contribute your changes to the project
title: '[pull-request]'
labels: ''
assignees: ''
---
**Description**
alw-integration library plus spring-boot-starter plus example-app finished

**Reference**
Issues #189 #190 #191

**Checklist**

- [x] Acceptance criteria are met
- [ ] Pipeline has been run successfully
- [ ] JUnit tests have been run successfully
- [ ] Changelog is adjusted
- [x] [Documentations](https://git.muenchen.de/groups/digitalisierung/-/wikis/Dokumentationen) are completed
- [ ] Frontend is tested
- [x] Created sub-branches are deleted
- [x] Boards are updated ( [Digitalisierung](https://git.muenchen.de/groups/digitalisierung/-/boards) , [Support](https://git.muenchen.de/digitalisierung/digiwf-support) , [Zenhub](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) )
- [ ] feature process is created or extended
- [ ] [git-ops](https://git.muenchen.de/digitalisierung/digiwf-ops) is customized
- [ ] Openshift environments are prepared (Secrets, etc.)

### CODE REVIEW

File: digiwf-alw-integration-starter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            <artifactId>spring-boot-starter</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
+
+        <!-- streaming -->
+        <dependency>
+            <groupId>io.muenchendigital.digiwf</groupId>
+            <artifactId>digiwf-spring-cloudstream-utils-starter</artifactId>
+            <version>0.0.1</version>
+        </dependency>
```
Comment (lmoesle) [inline digiwf-alw-integration-starter/pom.xml:47]: Do you need this dependency in the starter module?

File: digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/configuration/ConfigurationGuard.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    private SachbearbeitungMapper mapper;
+
+    @Override
+    public void afterPropertiesSet() {
+        if (!mapper.isInitialized()) {
+            final String message = "alw-sachbearbeitung.properties is empty";
+            log.error(message);
+            throw new IllegalArgumentException(message);
+        }
+    }
```
Comment (lmoesle) [inline digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/configuration/ConfigurationGuard.java:30]: Another option could be to use the `@PostConstruct`  annotation to execute the `afterPropertiesSet` method after construction. With this approach you don't have to implement the `InitializingBean` interface.

File: digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/infrastructure/streaming/MessageProcessor.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c): it@M - Dienstleister für Informations- und Telekommunikationstechnik
+ * der Landeshauptstadt München, 2020
+ */
+package io.muenchendigital.digiwf.alw.integration.infrastructure.streaming;
```
Comment (lmoesle) [inline digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/infrastructure/streaming/MessageProcessor.java]: I usually place the configs for spring cloud stream consumers and producers in the api package. They are kinda similar endpoints to rest controllers.

**Review by lmoesle — APPROVED** [review]
@martind260 👍🏻  The starter looks good to me. I only have a few small comments and I added some more information to the README.md (like e.g. shields).

File: digiwf-alw-integration-starter/pom.xml (hunk 1 above)
Comment (martind260) [reply digiwf-alw-integration-starter/pom.xml:47]: Actually yes, but just the component scan for the service beans.

File: digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/infrastructure/streaming/MessageProcessor.java (hunk 3 above)
Comment (martind260) [reply digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/infrastructure/streaming/MessageProcessor.java]: OK, youre right. Done

File: digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/configuration/ConfigurationGuard.java (hunk 2 above)
Comment (martind260) [reply digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/configuration/ConfigurationGuard.java:30]: What is the advantage of using @PostConstruct? The interface doesnt hurt. ;-)

Comment (lmoesle) [reply digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/configuration/ConfigurationGuard.java:30]: I don't see any advantages or disadvantages. It's just an alternative implementation and  I'm very used to the implementation with `@PostConstruct` . At miragon we usually use `@PostConstruct` but your implementation is totally fine 👍🏻

**Pull request merged**